shell/platform/common/client_wrapper/basic_message_channel_unittests.cc

```diff
@@ -86,4 +103,57 @@ TEST(BasicMessageChannelTest, Unregistration) {
   EXPECT_EQ(messenger.last_message_handler(), nullptr);
 }
 
+// Tests that calling Resize generates the binary message expected by the Dart
+// implementation.
+TEST(BasicMessageChannelTest, Resize) {
+  TestBinaryMessenger messenger;
+  const std::string channel_name("flutter/test");
+  BasicMessageChannel channel(&messenger, channel_name,
+                              &flutter::StandardMessageCodec::GetInstance());
+
+  channel.Resize(3);
+
+  // The expected content was created from the following Dart code:
+  //   MethodCall call = MethodCall('resize', ['flutter/test',3]);
+  //   StandardMethodCodec().encodeMethodCall(call).buffer.asUint8List();
+  const int expected_message_size = 29;
+
+  EXPECT_EQ(messenger.send_called(), true);
+  EXPECT_EQ(messenger.last_message_size(), expected_message_size);
+
+  int expected[expected_message_size] = {
```

Comment:
Could we use `StandardMessageCodec::DecodeMessage` to deserialize to an `EncodableValue`, and then assert the `EncodableValue` is expected? Debugging bytes can be painful

Reply:
I agree that relying on EncodableValue would have been more readable.
The first reason I choosed the bytes approach is that this is the way it is tested on Dart side.
And because the Dart side relies on its own deserializing logic, it turned out that testing based on the bytes array was helpful to catch potential issues due to int conversion (see https://github.com/flutter/engine/pull/44848#discussion_r1315919306).

Reply:
As in the linked case, we should have a comment in the code here explaining why it's tested this way (i.e., why it's critical to generate these exact bytes).